<?xml version="1.0"?>
<launch>
  <arg name="namespace" default="/navigation"/>
  <arg name="cmd_timeout" default="0.5"/>
  <arg name="starting_x" default="0"/>
  <arg name="starting_y" default="0"/>

  <!-- Vehicle pose -->
  <arg name="ackermann_cmd_topic" default="/vehicle/ackermann_cmd" />
  <arg name="odom_topic" default="/vehicle/odom" />
  <arg name="cmd_vel_topic" default="/vehicle/cmd_vel" />
  <arg name="new_wp_topic" default="/waypoints/goal" />

  <group ns="$(arg namespace)">
    <!-- Convert cmd_vel to ackermann drive -->
    <node name="follow_waypoints" pkg="ucsd_f1tenth_navigation_v2"
          type="follow_waypoints.py" >
      <param name="ackermann_cmd_topic" value="$(arg ackermann_cmd_topic)" />
      <param name="odom_topic" value="$(arg odom_topic)" />
      <param name="cmd_vel_topic" value="$(arg cmd_vel_topic)" />
      <param name="new_wp_topic" value="$(arg new_wp_topic)" />
      <param name="starting_x" value="$(arg starting_x)" />
      <param name="starting_y" value="$(arg starting_y)" />
    </node>
  </group>
</launch>
